erpnext/stock/doctype/stock_entry/stock_entry.js

```diff
@@ -238,6 +238,13 @@ frappe.ui.form.on('Stock Entry', {
 			}, __("Get items from"));
 		}
 
+		if (frm.doc.docstatus === 0 &&
+			['Material Issue', 'Material Receipt', 'Material Transfer', 'Send to Subcontractor'].includes(frm.doc.purpose)) {
+				frm.add_custom_button(__('Bill of Materials'), function() {
+					frm.events.get_items_from_bom(frm);
+				}, __("Get items from"));
+		}
```

Comment:
Code is repeated, create a separate method and call it from refresh and stock_entry_type trigger

Reply:
done